src/plugins/part.c

```diff
@@ -984,6 +984,171 @@ gboolean bd_part_delete_part (const gchar *disk, const gchar *part, GError **err
+    if (!ped_geometry_set_start (geom, ped_part->geom.start)) {
+        set_parted_error (error, BD_PART_ERROR_FAIL);
+        g_prefix_error (error, "Failed to set original start of geometry for partition '%d' on device '%s'", part_num, disk);
+        ped_constraint_destroy (constr);
+        ped_geometry_destroy (geom);
+        bd_utils_report_finished (progress_id, (*error)->message);
+        return FALSE;
+    }
+    if (size == 0) {
+        new_size = geom->length;
+    } else {
+        new_size = (size + dev->sector_size - 1) / dev->sector_size;
+    }
+
+    ped_geometry_destroy (geom);
+    geom = ped_geometry_new (dev, (PedSector) ped_part->geom.start, (PedSector) new_size);
+    if (!geom) {
+        set_parted_error (error, BD_PART_ERROR_FAIL);
+        g_prefix_error (error, "Failed to create geometry for partition '%d' on device '%s'", part_num, disk);
+        ped_constraint_destroy (constr);
+        bd_utils_report_finished (progress_id, (*error)->message);
+        return FALSE;
+    }
+
+    if (size != 0 && !ped_geometry_set_end (geom, ped_alignment_align_up (constr->end_align, constr->end_range, (PedSector) geom->end))) {
+        set_parted_error (error, BD_PART_ERROR_FAIL);
+        g_prefix_error (error, "Failed to set end of geometry for partition '%d' on device '%s'", part_num, disk);
+        ped_constraint_destroy (constr);
+        ped_geometry_destroy (geom);
+        bd_utils_report_finished (progress_id, (*error)->message);
+        return FALSE;
+    }
+
+    ped_constraint_destroy (constr);
+    constr = ped_constraint_exact (geom);
+    status = ped_disk_set_partition_geom (ped_disk, ped_part, constr, geom->start, geom->end);
+
+    if (status == 0 || ped_part->geom.start != orig_start || ped_part->geom.length < new_size) {
+        set_parted_error (error, BD_PART_ERROR_FAIL);
+        if (ped_part->geom.start != orig_start)
+            g_prefix_error (error, "Failed to meet original start of partition '%d' on device '%s'", part_num, disk);
+        else if (ped_part->geom.length < new_size)
+            g_prefix_error (error, "Failed to reach requested size of partition '%d' on device '%s'", part_num, disk);
+        else
+            g_prefix_error (error, "Failed to resize partition '%d' on device '%s'", part_num, disk);
+        ped_disk_destroy (ped_disk);
+        ped_device_destroy (dev);
+        ped_geometry_destroy (geom);
+        bd_utils_report_finished (progress_id, (*error)->message);
+        return FALSE;
+    }
+
+    ret = disk_commit (ped_disk, disk, error);
+
+    if (orig_flag_state != 0 && ped_disk_is_flag_available (ped_disk, PED_DISK_CYLINDER_ALIGNMENT)) {
+        ped_disk_set_flag (ped_disk, PED_DISK_CYLINDER_ALIGNMENT, orig_flag_state);
+    }
+
+    ped_disk_destroy (ped_disk);
+    ped_device_destroy (dev);
+    ped_geometry_destroy (geom);
+    bd_utils_report_finished (progress_id, "Completed");
+
+    return ret;
+}
```

Comment:
It was not possible for me to find a proper solution that way and updating the geometry is I think more reliable to preserve the partition properties than copying flags and labels.
Also the way add_part_to_disk added maximum sized partitions was problematic and buggy if a maximal partition should be placed in front of existing one (all my tries to fix it still failed my test_create_resize_part_two).
From my point of view this is corrected in a good way now by using ped_disk_get_max_partition_geometry which requires an existing partition, so add_part_to_disk makes a new partition with temporary size and then calls the resize function. That way the partition is guaranteed to have the requested start position (plus possible alignment in only one way during creation) and at least the requested size.
(I chose 1 MiB as initial size for partitions as needed for logical partitions.)